### ISSUE TITLE
Bug using Fig with an int

### ISSUE THREAD

**Issue by gouarin** [vispy/vispy#1095]
Hi,

it's not possible to initiate a Fig using an integer whereas it should be possible if we look at the source code.

The problem is in grid.py in __getitem__ method when the size of the index is one. Add Slice(None) doesn't work if there is no other widget because the grid_size in y is 0.

Slice(0, 1, None) by default could solve the problem but it's probably not the best way because if a user do 

fig = Fig()
ax = fig[0]
ax.image ...

ax2 = fig[1, 1:3]
...
Probably, the user want that the first plot take all the columns and not just one ...

Best,
Loic

**Comment by Eric89GXL** [comment]
Yeah, the accessing API could probably be improved. PR welcome :)
